instrumentation/dubbo/dubbo-apache-2.7/library/src/main/java/io/opentelemetry/instrumentation/dubbo/apache/v2_7/OpenTelemetryFilter.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.dubbo.apache.v2_7;
+
+import static io.opentelemetry.api.trace.Span.Kind.CLIENT;
+import static io.opentelemetry.api.trace.Span.Kind.SERVER;
+import static io.opentelemetry.instrumentation.dubbo.apache.v2_7.DubboInjectAdapter.SETTER;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Span.Kind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import java.util.concurrent.CompletableFuture;
+import org.apache.dubbo.common.extension.Activate;
+import org.apache.dubbo.rpc.Filter;
+import org.apache.dubbo.rpc.Invocation;
+import org.apache.dubbo.rpc.Invoker;
+import org.apache.dubbo.rpc.Result;
+import org.apache.dubbo.rpc.RpcContext;
+import org.apache.dubbo.rpc.RpcException;
+import org.apache.dubbo.rpc.RpcInvocation;
+
+@Activate(group = {"consumer", "provider"})
+public class OpenTelemetryFilter implements Filter {
+  private final DubboTracer tracer;
+
+  public OpenTelemetryFilter() {
+    this.tracer = new DubboTracer();
+  }
+
+  @Override
+  public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException {
+    if (!(invocation instanceof RpcInvocation)) {
+      return invoker.invoke(invocation);
+    }
+    String methodName = invocation.getMethodName();
+    String interfaceName = invoker.getInterface().getName();
+    RpcContext rpcContext = RpcContext.getContext();
+    Kind kind = rpcContext.isProviderSide() ? SERVER : CLIENT;
+    final Context context;
+    if (kind.equals(CLIENT)) {
+      context = tracer.startClientSpan(interfaceName, methodName);
+      GlobalOpenTelemetry.getPropagators()
+          .getTextMapPropagator()
+          .inject(context, (RpcInvocation) invocation, SETTER);
```

Comment:
Not for this PR, nor necessarily you, but `inject` needs to be moved from `HttpClientTracer` to `BaseTracer`